app/src/main/java/org/dslul/openboard/inputmethod/latin/LatinIME.java

```diff
@@ -1336,6 +1361,119 @@ public void updateFullscreenMode() {
+                                .setTextColor(Color.parseColor("#FF202124"))
+                                .setTextSize(12)
+                                .build())
+                .setSubtitleStyle(
+                        new TextViewStyle.Builder()
+                                .setLayoutMargin(0, 0, toPixel(4), 0)
+                                .setTextColor(Color.parseColor("#99202124"))
+                                .setTextSize(10)
+                                .build())
+                .setEndIconStyle(new ImageViewStyle.Builder().setLayoutMargin(0, 0, 0, 0).build())
+                .build();
+        stylesBuilder.addStyle(style);
+
+        Bundle stylesBundle = stylesBuilder.build();
+
+        final ArrayList<InlinePresentationSpec> presentationSpecs = new ArrayList<>();
+        presentationSpecs.add(new InlinePresentationSpec.Builder(new Size(100, getHeight()),
+                new Size(740, getHeight())).setStyle(stylesBundle).build());
+        presentationSpecs.add(new InlinePresentationSpec.Builder(new Size(100, getHeight()),
+                new Size(740, getHeight())).setStyle(stylesBundle).build());
+
+        return new InlineSuggestionsRequest.Builder(presentationSpecs)
+                .setMaxSuggestionCount(6)
+                .build();
+    }
+
+    private int toPixel(int dp) {
+        return (int) TypedValue.applyDimension(COMPLEX_UNIT_DIP, dp,
+                getResources().getDisplayMetrics());
+    }
+
+    private int getHeight() {
+        return getResources().getDimensionPixelSize(R.dimen.config_suggestions_strip_height);
+    }
+
+    @Override
+    @RequiresApi(api = Build.VERSION_CODES.R)
+    public boolean onInlineSuggestionsResponse(InlineSuggestionsResponse response) {
+        Log.d(TAG,"onInlineSuggestionsResponse called");
+
+        final List<InlineSuggestion> inlineSuggestions = response.getInlineSuggestions();
+
+        if (inlineSuggestions.isEmpty()) {
+            return false;
+        }
+
+        final int totalSuggestionsCount = inlineSuggestions.size();
+
+        // A container to hold all views
+        LinearLayout container = new LinearLayout(mDisplayContext);
+
+        for (int i = 0; i < totalSuggestionsCount; i++) {
+            final InlineSuggestion inlineSuggestion = inlineSuggestions.get(i);
+
+            inlineSuggestion.inflate(this, new Size(ViewGroup.LayoutParams.WRAP_CONTENT,
+                    ViewGroup.LayoutParams.WRAP_CONTENT), getMainExecutor(), (view) -> {
+                if (view != null)
+                    container.addView(view);
+            });
+        }
+
+        HorizontalScrollView view = new HorizontalScrollView(mDisplayContext);
+        view.setLayoutParams(new ViewGroup.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
+        view.setHorizontalScrollBarEnabled(false);
+
+        view.addView(container);
+
+        // Delay required to show properly
```

Comment:
What is happening without delay?

Reply:
The suggestions only appear for a split second.

Reply:
Ah, that probably comes from a suggestion update right after.
Could you try whether `mHandler.cancelUpdateSuggestionStrip()` helps instead of a delay?

Reply:
It is not working. Still the same issue.

Reply:
Alright, then let's leave the delay